### DISCUSSION TITLE
Auth0Provider - How to direct to signup instead of login page

### DISCUSSION THREAD

**Discussion by shahbhavik01** [nextauthjs/next-auth#7164] [Help] [answered]
### Question 💬

I'm creating a next-auth SignIn and SignUp page with Auth0 provider. When the user clicks "Sign In", they are directed to a URL like "https://MY-AUTH0ISSUER.us.auth0.com/u/login?state=SOMESECRET". 

When a user clicks "Sign Up", they're still directed to the same URL. I want the users to be directed to "https://MY-AUTH0ISSUER.us.auth0.com/u/signup?state=SOMESECRET" instead. How can I achieve this? 

### How to reproduce ☕️

My [...nextauth].ts is setup with the Auth0Provider: 

  providers: [
    Auth0Provider({
      clientId: process.env.AUTH0_CLIENT_ID || "",
      clientSecret: process.env.AUTH0_CLIENT_SECRET || "",
      issuer: process.env.AUTH0_ISSUER,
    }),
  ],
  
  
  I'm passing the same onClick function to both Sign In and Sign Up. 

      onClick={() =>
        signIn(provider.id, {
          callbackUrl: process.env.VERCEL_URL || "http://localhost:3000",
        })
      }

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by balazsorban44** [comment]
You have multiple options. 


Yocan do this by using the advanced initialization and modifying the `authorization` url property on the `Auth0` provider.

https://next-auth.js.org/configuration/initialization#advanced-initialization

https://next-auth.js.org/configuration/providers/oauth#override-default-options


Or just add two Auth0 providers, where the second has a different id (eg `auth0-register` that you call with `signIn("auth0-register")` and authorization url.

**Comment by TCatshoek** [comment, marked as answer]
If I'm not mistaken, with auth0 you can also do this by passing additional parameters to the signIn function (which are added to the call to the `/authorize` endpoint)

https://auth0.com/docs/authenticate/login/auth0-universal-login/new-experience#signup

So it would look something like:
```
signIn("auth0", undefined, {"screen_hint": "signup"})
```

I haven't tested this as I haven't used auth0 myself, but I stumbled upon this working on a similar issue with amazon cognito. Cognito does not support the extra parameters auth0 does so I settled on doing it the way @balazsorban44 suggested.

**Reply by shahbhavik01** [reply]
Thanks @TCatshoek  That was it! This does it. 

`      onClick={() =>
        signIn(provider.id, {
          callbackUrl: process.env.VERCEL_URL || "http://localhost:3000",
        }, { "screen_hint": "signup" })
      }
`

**Reply by theis-aidia** [reply]
Awesome, thanks. This was extremely hard to find 🙌
